### PR TITLE
Remove GetEffectiveFlowDirection and defer flow direction handling to platforms

### LINKED ISSUE #8175: Performance issue in LayoutExtensions.ShouldArrangeLeftToRight
### Description

A chunk of time is spent in `LayoutExtensions.ShouldArrangeLeftToRight()`:

image

[maui-app_20220620_085602.speedscope.json.zip](https://github.com/dotnet/maui/files/8941918/maui-app_20220620_085602.speedscope.json.zip)

It appears the default behavior is for every MAUI view, to recurse all the way to the root `View` of the app:

image

https://github.com/dotnet/maui/blob/c2c64e5a990e51fc19c209a2a3b6e4257276ffe5/src/Core/src/Layouts/LayoutExtensions.cs#L210-L222

https://github.com/dotnet/maui/blob/c2c64e5a990e51fc19c209a2a3b6e4257276ffe5/src/Core/src/Core/Extensions/IViewExtensions.cs#L5-L21

So let's say you have a nested layout, 5 views deep, with 100 `Label` on the screen. Each `Label` we end up querying the same views for flow direction 500 times.

Is there a way the default behavior could avoid recursion through parent views?

### Steps to Reproduce

1. Run this sample: https://github.com/unoplatform/performance/tree/master/src/dopes/DopeTestMaui
2. Click the button that says `Build` when the app launches
3. Record a `.speedscope` or `.nettrace` on Android: https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/tracing.md

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows, macOS

### Affected platform versions

Any

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change

Automates reversal of RTL layouts during platform arrange for iOS and Android (basically works like Windows now).
Removes all cross-platform RTL layout code (since the platforms handle it). Updates automated tests accordingly.
Propagates iOS LTR/RTL settings when FlowDirection is MatchParent (this wasn't happening at all before).
Consolidates code to update text alignment on iOS when FlowDirection changes.
Adds FlowDirection propagation tests for iOS (skips the other platforms because they do this automatically).

### Issues Fixed

Fixes #8175 (in that it removes `ShouldArrangeLeftToRight` entirely).

This is an alternative to #9326/#9327.

### CODE REVIEW

**Comment by hartez** [comment]
Leaving this as draft until I can get around to updating all the public API files and removing the now-unused extension methods.

File: src/Core/src/Handlers/Entry/EntryHandler.cs [hunk 1]
```diff
@@ -45,9 +45,6 @@ public partial class EntryHandler : IEntryHandler
 
 		static EntryHandler()
```
Comment (PureWeen) [inline src/Core/src/Handlers/Entry/EntryHandler.cs:46]: Can we just delete the entire static ctor now?

**Comment by PureWeen** [comment]
@hartez  the arranging on CollectionView looks to be better now but the sizing seems off now in RTL mode

The Left image is this PR
The Right image is without this PR

image

**Comment by hartez** [comment]
Hmmm... the problems on Android in the CollectionView go away if I remove the Border from the template.

**Comment by hartez** [comment]
@PureWeen The Android problems should be fixed - looking at the iOS issues now.

**Review by PureWeen — APPROVED** [review]
https://github.com/dotnet/maui/issues/9701

https://github.com/dotnet/maui/issues/9702

**Comment by PureWeen** [comment]
https://github.com/dotnet/maui/issues/9701
https://github.com/dotnet/maui/issues/9702

**Pull request merged**